pkg/api/meta/restmapper.go

```diff
@@ -148,6 +148,23 @@ func kindToResource(kind string, mixedCase bool) (plural, singular string) {
 	return
 }
 
+// ResourceSingularlizer converts a resource name from plural to singular (e.g., from pods to pod)
+func ResourceSingularlizer(resource string) (singular string) {
```

Comment:
Thanks for the suggestion. Change this to be a method on the interface RESTMapper.